src/sharing/components/ContactSuggestion.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import get from 'lodash/get'
+import { Avatar } from 'cozy-ui/react'
+import { translate } from 'cozy-ui/react/I18n'
+
+import styles from './recipient.styl'
+
+import { getPrimaryEmail, getPrimaryCozy, getDisplayName } from '..'
+import { DOCTYPE_GROUPS } from '../../drive/lib/doctypes'
+import Identity from 'sharing/components/Identity'
+import { getInitials } from './Recipient'
+
+export const ContactSuggestion = ({ contactOrGroup, contacts, t }) => {
+  let avatarText, name, details
+  if (contactOrGroup._type === DOCTYPE_GROUPS) {
+    name = contactOrGroup.name
+    const membersCount = contacts
+      .reduce((total, contact) => {
+        if (
+          get(contact, 'relationships.groups.data', [])
+            .map(group => group._id)
+            .includes(contactOrGroup._id)
+        ) {
+          return total + 1
+        }
+
+        return total
+      }, 0)
+      .toString()
+    details = t('Share.members.count', {
+      smart_count: membersCount
+    })
+    avatarText = 'G'
+  } else {
+    name = getPrimaryEmail(contactOrGroup)
+    if (name === undefined) {
+      name = getDisplayName(contactOrGroup)
+    }
+    avatarText = getInitials(name)
+    details = getPrimaryCozy(contactOrGroup)
+  }
+  return (
+    <div className={styles['recipient']}>
+      <Avatar text={avatarText} size="small" />
+      <Identity name={name} details={details} />
+    </div>
+  )
+}
```

Comment:
I'm wondering if we should not divide the logic and the SuggestionComponent. It'll be easier to test the name / details / reduce stuff?